polyglot/piranha/src/models/source_code_unit.rs

```diff
@@ -130,7 +142,9 @@ impl SourceCodeUnit {
         return comment_range;
       }
     } else if buffer > 0 {
-      return self.get_comment_at_line(row - 1, buffer - 1);
+      // We pass [start_byte] itself, because we know that parent of the current row is the parent of the above row too.
+      // If that's not the case, its okay, because we will not find any comments in these scenarios.
```

Comment:
Maybe worth having a sanity test case such as:

```
// comment is not removed
if (someCond()) { callToBeRemoved(); }
```

?

Reply:
I will add this to the test case.